cmd/k8sagent/main_nix.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2012-2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/juju/clock"
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/featureflag"
+	"github.com/juju/loggo"
+	proxyutils "github.com/juju/proxy"
+
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/jujud/dumplogs"
+	"github.com/juju/juju/cmd/jujud/introspect"
+	"github.com/juju/juju/cmd/jujud/run"
+	cmdutil "github.com/juju/juju/cmd/jujud/util"
+	initcommand "github.com/juju/juju/cmd/k8sagent/init"
+	unitcommand "github.com/juju/juju/cmd/k8sagent/unit"
+	components "github.com/juju/juju/component/all"
+	"github.com/juju/juju/core/machinelock"
+	"github.com/juju/juju/juju/names"
+	"github.com/juju/juju/juju/osenv"
+	"github.com/juju/juju/utils/proxy"
+	"github.com/juju/juju/worker/logsender"
+)
+
+var logger = loggo.GetLogger("juju.cmd.k8sagent")
+
+func init() {
+	featureflag.SetFlagsFromEnvironment(osenv.JujuFeatureFlagEnvKey)
```

Comment:
I think we probably want to remove this. This approach was from way way back, and we are trying to move to controller feature flags now.